src/main/java/org/example/api/DeliveryEmployeeService.java

```diff
@@ -53,4 +62,21 @@ public DeliveryEmployee getDeliveryEmployeeById(int id) throws DoesNotExistExcep
 
         return deliveryEmployee;
     }
+
+    public void updateDeliveryEmployee(int id, UpdateDeliveryEmployeeRequest deliveryEmployeeUpdate) throws FailedToUpdateDeliveryEmployee, FailedToGetException, DeliveryEmployeeDoesNotExistException, InvalidUpdateRequestException {
+
+        // call to dao to check employee with given id is a delivery employee
+        if(dao.checkEmployeeIsDeliveryEmployee(id) == null){
+            throw new DeliveryEmployeeDoesNotExistException();
+        }
+
+        // if employee exists, check data is valid using employee validator
+        UpdateDeliveryEmployeeValidator updateDeliveryEmployeeValidator = new UpdateDeliveryEmployeeValidator();
+        if(updateDeliveryEmployeeValidator.isValidEmployeeUpdate(deliveryEmployeeUpdate) != null){
+            throw new InvalidUpdateRequestException(updateDeliveryEmployeeValidator.isValidEmployeeUpdate(deliveryEmployeeUpdate));
```

Comment:
rather than calling the same method 2 times, just assign it to a variable and use it 2 times